src/parser/jobs/mch/modules/Reassemble.tsx

```diff
@@ -1,86 +1,108 @@
-import {Trans, Plural} from '@lingui/react'
-import {ActionLink} from 'components/ui/DbLink'
-import {ActionRoot} from 'data/ACTIONS/root'
-import STATUSES from 'data/STATUSES'
-import {CastEvent, BuffEvent} from 'fflogs'
-import Module, {dependency} from 'parser/core/Module'
-import Combatants from 'parser/core/modules/Combatants'
+import {Plural, Trans} from '@lingui/react'
+import {DataLink} from 'components/ui/DbLink'
+import {ActionKey} from 'data/ACTIONS'
+import {Event, Events} from 'event'
+import {Analyser} from 'parser/core/Analyser'
+import {filter, oneOf} from 'parser/core/filter'
+import {dependency} from 'parser/core/Injectable'
 import {Data} from 'parser/core/modules/Data'
-import Suggestions, {TieredSuggestion, SEVERITY} from 'parser/core/modules/Suggestions'
+import Suggestions, {SEVERITY, TieredSuggestion} from 'parser/core/modules/Suggestions'
 import React from 'react'
 
 // These are the only GCDs that should be reassembled under normal circumstances
-const REASSEMBLE_GCDS: Array<keyof ActionRoot> = [
+const REASSEMBLE_GCDS: ActionKey[] = [
 	'AIR_ANCHOR',
 	'AUTO_CROSSBOW',
 	'DRILL',
 	'SPREAD_SHOT',
+	// 6.0:
+	//'CHAIN_SAW',
+	//'SCATTER_GUN',
 ]
 
-const OTHER_GCDS: Array<keyof ActionRoot> = [
+const OTHER_GCDS: ActionKey[] = [
 	'BIOBLASTER',
 	'HEAT_BLAST',
 	'HEATED_SPLIT_SHOT',
 	'HEATED_SLUG_SHOT',
 	'HEATED_CLEAN_SHOT',
 ]
 
-export default class Reassemble extends Module {
+export class Reassemble extends Analyser {
 	static override handle = 'reassemble'
 
-	@dependency private combatants!: Combatants
 	@dependency private data!: Data
 	@dependency private suggestions!: Suggestions
 
 	private badReassembles = 0
 	private droppedReassembles = 0
-	private lastUse = 0
+	private lastGcd = 0
+	private reassembleActive = false
 
 	private REASSEMBLE_GCDS: number[] = []
 	private OTHER_GCDS: number[] = []
 
-	protected override init() {
+	override initialise() {
 		this.REASSEMBLE_GCDS = REASSEMBLE_GCDS.map(key => this.data.actions[key].id)
 		this.OTHER_GCDS = OTHER_GCDS.map(key => this.data.actions[key].id)
```

Comment:
Not actually part of the diff, but prefer camel-casing here so it's clearer if you're using the const or the local var, and also I'm impressed linter didn't rage at this tbh.